src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,42 @@
+package duke;
+
+import duke.command.Command;
+import duke.task.TaskList;
+
+public class Duke {
+    private Storage storage;
+    private TaskList taskList;
+    private Ui ui;
+
+    public Duke(String filepath) {
+        try {
+            this.storage = new Storage(filepath);
+            this.taskList = new TaskList(storage.loadTasks());
+            this.ui = new Ui();
+        } catch (DukeException err) {
+            System.out.println(err.getMessage());
+        }
+    }
+
+    public void run() {
+        ui.showWelcomeMessage();
+        boolean isExit = false;
```

Comment:
The boolean variable isExit explains the named entity to the reader accurately and at a sufficient level of detail.